app/models/user.rb

```diff
@@ -0,0 +1,36 @@
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+  has_many :images, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :commented_images, through: :comments, source: :image
+  validates :username,
+  :presence => true,
+  :uniqueness => {
```

Comment:
Style/HashSyntax: Use the new Ruby 1.9 hash syntax. (https://github.com/bbatsov/ruby-style-guide#hash-literals)